src/components/EditorMarkdownIt.vue

```diff
@@ -30,13 +37,46 @@ export default {
 	},
 
 	created() {
+		this.setImageRule(this.noteid)
 		this.onUpdate()
 	},
 
 	methods: {
 		onUpdate() {
 			this.html = this.md.render(this.value)
 		},
+		setImageRule(id) {
+			// https://github.com/markdown-it/markdown-it/blob/master/docs/architecture.md#renderer
+			// Remember old renderer, if overridden, or proxy to default renderer
+			const defaultRender = this.md.renderer.rules.image || function(tokens, idx, options, env, self) {
+				return self.renderToken(tokens, idx, options)
+			}
+
+			this.md.renderer.rules.image = function(tokens, idx, options, env, self) {
+				// If you are sure other plugins can't add `target` - drop check below
+				const token = tokens[idx]
+				const aIndex = token.attrIndex('src')
+				let path = token.attrs[aIndex][1]
+
+				if (!path.startsWith('http')) {
+					path = generateUrl('apps/notes') + '/notes/' + id + '/attachment?path=' + path
+				}
+
+				token.attrs[aIndex][1] = path
+				// pass token to default renderer.
+				if (path.endsWith("jpg") ||
+					path.endsWith("jpeg") ||
+					path.endsWith("bmp") ||
+					path.endsWith("webp") ||
+					path.endsWith("gif") ||
+					path.endsWith("png")) {
```

Comment:
The check should ignore the letter case, e.g. it should also work with `JPG` instead of `jpg`.